src/providers/aws.rs

```diff
@@ -1,3 +1,29 @@
+//! AWS backend for tsunami.
+//!
+//! The primary `impl Launcher` type is [`AWSLauncher`].
+//! It internally uses the lower-level, region-specific [`AWSRegion`].
+//! Both these types use [`MachineSetup`] as their descriptor type.
+//!
+//! This implementation uses [defined duration](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances)
+//! instances.
+//!
+//! # Example
+//! ```rust,no_run
+//! use rusoto_core::Region;
+//! use slog::info;
+//! use tsunami::providers::{aws, Launcher};
+//! use tsunami::{Machine, TsunamiBuilder};
+//!
+//! let mut b = TsunamiBuilder::default();
+//! b.add("my machine", aws::MachineSetup::default()).unwrap();
+//! let mut l: tsunami::providers::aws::AWSLauncher = Default::default();
```

Comment:
`aws` is already in scope, so this can just be `aws::AWSLauncher`.

Reply:
Speaking of, I wonder if `AWSLauncher` should be renamed to `Launcher` given that it's already under `mod aws`?

Reply:
Also, this might look cleaner as:
```rust
let mut l = aws::AWSLauncher::default();
```

Reply:
yes, and `aws::MachineSetup` should just be `aws::Setup`, to match with the others.
the `AWSLauncher` comment also applies to `AzureLauncher`